compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -141,6 +119,10 @@ pub(crate) fn finalize(cx: &CodegenCx<'_, '_>) {
         llvm::set_linkage(array, llvm::Linkage::InternalLinkage);
         llvm::set_initializer(array, initializer);
     }
+
+    // Generate the coverage map header, which contains the filenames used by
+    // this CGU's coverage mappings, and store it in a well-known global.
+    generate_covmap_record(cx, covmap_version, filenames_size, filenames_val);
```

Comment:
This is deliberate.

If the object file contains a covmap record but doesn’t have any covfun records, the `llvm-cov` tool will fail when generating a coverage report.

And if there are no covfun records then the covmap record is useless anyway, so omitting it is correct.

Reply:
Could you add your elaboration as a comment, as this might not be immediately obvious?

Reply:
It's mentioned in the comment for the early return, but I'll add a note to the `generate_covmap_record` call as well, because it's not obvious in that place.